src/blockdata/transaction.rs

```diff
@@ -220,12 +220,104 @@ impl Default for TxIn {
         TxIn {
             previous_output: OutPoint::default(),
             script_sig: Script::new(),
-            sequence: u32::max_value(),
+            sequence: Sequence::MAX,
             witness: Witness::default(),
         }
     }
 }
 
+/// Bitcoin transaction input sequencing number.
+#[derive(Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Debug, Hash)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+#[cfg_attr(feature = "serde", serde(crate = "actual_serde"))]
+pub struct Sequence(pub u32);
+
+impl Sequence {
+    /// The maximum allowable sequence number
+    pub const MAX: Self = Sequence(MAX_SEQUENCE);
+    /// The lowest sequence number that does not opt-in for replace-by-fee.
+    pub const RBF_SIGNAL: Self = Sequence(0xFFFFFFFE);
+    /// Zero value sequence
+    pub const ZERO: Self = Sequence(0);
+    /// BIP-68 relative time lock disable flag mask
+    const DISABLE_FLAG_MASK: u32 = 0x80000000;
+    /// BIP-68 relative time lock type flag mask
+    const LOCK_TYPE_MASK: u32 = 0x00400000;
+    
+    /// Retuns `true` if the sequence number is 0xffffffff.
+    #[inline]
+    pub fn is_final(&self) -> bool {
+        *self == Sequence::MAX
+    }
+
+    /// Returns `true` if the sequence number is less than 0xfffffffe which
+    /// indicates that the transaction is opted-in to BIP125 replace-by-fee.
+    #[inline]
+    pub fn is_rbf(&self) -> bool {
+        *self < Sequence::RBF_SIGNAL
+    }
+
+    /// Returns `true` if the sequence has a relative locktime.
+    #[inline]
+    pub fn is_relative_lock_time(&self) -> bool {
```

Comment:
I think @nlanson is a native English speaker but I'll chime in with 'is'. Also `s/locktime/lock time/` since we use undersore separation in `is_relative_lock_time`